airflow/providers/amazon/aws/operators/boto3.py

```diff
@@ -29,64 +30,65 @@
 
-    template_fields: Sequence[str] = ("boto3_kwargs",)
+    template_fields: Sequence[str] = ("client_type", "client_method", "method_kwargs",)
 
     def __init__(
         self,
-        *args,
-        aws_conn_id: str = "aws_conn_id",
-        hook_params: dict = None,
-        boto3_callable: str = None,
-        boto3_kwargs: dict = None,
+        aws_conn_id: str | None = AwsBaseHook.default_conn_name,
+        hook_params: dict | None = None,
+        client_type: str | None = None,
+        client_method: str | None = None,
+        method_kwargs: dict | None = None,
         **kwargs,
     ):
         """
         Args:
             aws_conn_id (str, optional): AWS connection id. Defaults to "aws_conn_id".
             hook_params (dict, optional): Parameters to pass to AwsBaseHook. Defaults to None.
-            boto3_callable (str, optional): boto3 function that the operator should call. Format: <client>.<function>. Example: "rds.describe_db_snapshots". Defaults to None.
-            boto3_kwargs (dict, optional): Parameters to pass to the boto3_callable. Example: {"DBInstanceIdentifier": "my-database"} Defaults to None.
+            client_type (str, optional): boto3 client. Ex: "rds", "ec2", "s3", etc. Defaults to None.
+            client_method (str, optional): boto3 function that the operator should call on the client. Example: "describe_db_snapshots" (with `client_type`="rds"). Defaults to None.
+            method_kwargs (dict, optional): Parameters to pass to the client_method. Example: {"DBInstanceIdentifier": "my-database"} Defaults to None.
         """
-        assert (
-            boto3_callable and len(boto3_callable.split(".")) == 2
-        ), f"Wrong boto3_callable: Got '{boto3_callable}' but expected format: '<boto3_client_type>.<function_to_call>'"
+        if not client_method.isidentifier():
+            raise ParamValidationError(f"\"{client_method}\" is not valid client_method")
+
+        self.client_type = client_type
+        self.client_method = client_method
+        self.method_kwargs = method_kwargs or {}
 
-        client_type, self.client_callable = boto3_callable.split(".")
-        self.boto3_kwargs = boto3_kwargs or {}
+        self.hook_params = hook_params or {}
 
-        hook_params = hook_params or {}
-        self.hook = AwsBaseHook(aws_conn_id=aws_conn_id, **hook_params, client_type=client_type)
-        super().__init__(*args, **kwargs)
+        super().__init__(**kwargs)
+
+    @cached_property
+    def hook(self) -> AwsBaseHook:
+        return AwsBaseHook(aws_conn_id=self.aws_conn_id, **self.hook_params, client_type=self.client_type)
 
     @property
     def boto3_action(
         self,
     ):
-        return getattr(self.hook.conn, self.client_callable)
+        method = getattr(self.hook.conn, self.client_method)
+        if not callable(method):
+            raise ParamValidationError(f"Method \"{self.client_method}\" does not exist on boto3 client \"{self.client_type}\"")
```

Comment:
psst... Do you want some kind of "undocumented feature" of `botocore` client 😄  ?
Client meta object (public property) has public property with mapping between `botocore` methods and AWS API methods.

```python
import boto3

client = boto3.client("sts")
print(str(client).center(60, "-"))
for botocore_method, api_method in client.meta.method_to_api_mapping.items():
    print(botocore_method, api_method)
print(f"Client has method 'get_caller_identity': {'get_caller_identity' in client.meta.method_to_api_mapping}")
print(f"Client has method 'add_role_to_db_cluster': {'add_role_to_db_cluster' in client.meta.method_to_api_mapping}")

client = boto3.client("rds", region_name="eu-west-1")
print(str(client).center(60, "-"))
for botocore_method, api_method in client.meta.method_to_api_mapping.items():
    print(botocore_method, api_method)
print(f"Client has method 'get_caller_identity': {'get_caller_identity' in client.meta.method_to_api_mapping}")
print(f"Client has method 'add_role_to_db_cluster': {'add_role_to_db_cluster' in client.meta.method_to_api_mapping}")
```

output
```
--------<botocore.client.STS object at 0x105f69af0>---------
assume_role AssumeRole
assume_role_with_saml AssumeRoleWithSAML
assume_role_with_web_identity AssumeRoleWithWebIdentity
decode_authorization_message DecodeAuthorizationMessage
get_access_key_info GetAccessKeyInfo
get_caller_identity GetCallerIdentity
get_federation_token GetFederationToken
get_session_token GetSessionToken
Client has method 'get_caller_identity': True
Client has method 'add_role_to_db_cluster': False
--------<botocore.client.RDS object at 0x106cc9f40>---------
add_role_to_db_cluster AddRoleToDBCluster
add_role_to_db_instance AddRoleToDBInstance
add_source_identifier_to_subscription AddSourceIdentifierToSubscription
add_tags_to_resource AddTagsToResource
apply_pending_maintenance_action ApplyPendingMaintenanceAction
authorize_db_security_group_ingress AuthorizeDBSecurityGroupIngress
backtrack_db_cluster BacktrackDBCluster
cancel_export_task CancelExportTask
copy_db_cluster_parameter_group CopyDBClusterParameterGroup
copy_db_cluster_snapshot CopyDBClusterSnapshot
copy_db_parameter_group CopyDBParameterGroup
copy_db_snapshot CopyDBSnapshot
copy_option_group CopyOptionGroup
create_custom_db_engine_version CreateCustomDBEngineVersion
create_db_cluster CreateDBCluster
create_db_cluster_endpoint CreateDBClusterEndpoint
create_db_cluster_parameter_group CreateDBClusterParameterGroup
create_db_cluster_snapshot CreateDBClusterSnapshot
create_db_instance CreateDBInstance
create_db_instance_read_replica CreateDBInstanceReadReplica
create_db_parameter_group CreateDBParameterGroup
create_db_proxy CreateDBProxy
create_db_proxy_endpoint CreateDBProxyEndpoint
create_db_security_group CreateDBSecurityGroup
create_db_snapshot CreateDBSnapshot
create_db_subnet_group CreateDBSubnetGroup
create_event_subscription CreateEventSubscription
create_global_cluster CreateGlobalCluster
create_option_group CreateOptionGroup
delete_custom_db_engine_version DeleteCustomDBEngineVersion
delete_db_cluster DeleteDBCluster
delete_db_cluster_endpoint DeleteDBClusterEndpoint
delete_db_cluster_parameter_group DeleteDBClusterParameterGroup
delete_db_cluster_snapshot DeleteDBClusterSnapshot
delete_db_instance DeleteDBInstance
delete_db_instance_automated_backup DeleteDBInstanceAutomatedBackup
delete_db_parameter_group DeleteDBParameterGroup
delete_db_proxy DeleteDBProxy
delete_db_proxy_endpoint DeleteDBProxyEndpoint
delete_db_security_group DeleteDBSecurityGroup
delete_db_snapshot DeleteDBSnapshot
delete_db_subnet_group DeleteDBSubnetGroup
delete_event_subscription DeleteEventSubscription
delete_global_cluster DeleteGlobalCluster
delete_option_group DeleteOptionGroup
deregister_db_proxy_targets DeregisterDBProxyTargets
describe_account_attributes DescribeAccountAttributes
describe_certificates DescribeCertificates
describe_db_cluster_backtracks DescribeDBClusterBacktracks
describe_db_cluster_endpoints DescribeDBClusterEndpoints
describe_db_cluster_parameter_groups DescribeDBClusterParameterGroups
describe_db_cluster_parameters DescribeDBClusterParameters
describe_db_cluster_snapshot_attributes DescribeDBClusterSnapshotAttributes
describe_db_cluster_snapshots DescribeDBClusterSnapshots
describe_db_clusters DescribeDBClusters
describe_db_engine_versions DescribeDBEngineVersions
describe_db_instance_automated_backups DescribeDBInstanceAutomatedBackups
describe_db_instances DescribeDBInstances
describe_db_log_files DescribeDBLogFiles
describe_db_parameter_groups DescribeDBParameterGroups
describe_db_parameters DescribeDBParameters
describe_db_proxies DescribeDBProxies
describe_db_proxy_endpoints DescribeDBProxyEndpoints
describe_db_proxy_target_groups DescribeDBProxyTargetGroups
describe_db_proxy_targets DescribeDBProxyTargets
describe_db_security_groups DescribeDBSecurityGroups
describe_db_snapshot_attributes DescribeDBSnapshotAttributes
describe_db_snapshots DescribeDBSnapshots
describe_db_subnet_groups DescribeDBSubnetGroups
describe_engine_default_cluster_parameters DescribeEngineDefaultClusterParameters
describe_engine_default_parameters DescribeEngineDefaultParameters
describe_event_categories DescribeEventCategories
describe_event_subscriptions DescribeEventSubscriptions
describe_events DescribeEvents
describe_export_tasks DescribeExportTasks
describe_global_clusters DescribeGlobalClusters
describe_option_group_options DescribeOptionGroupOptions
describe_option_groups DescribeOptionGroups
describe_orderable_db_instance_options DescribeOrderableDBInstanceOptions
describe_pending_maintenance_actions DescribePendingMaintenanceActions
describe_reserved_db_instances DescribeReservedDBInstances
describe_reserved_db_instances_offerings DescribeReservedDBInstancesOfferings
describe_source_regions DescribeSourceRegions
describe_valid_db_instance_modifications DescribeValidDBInstanceModifications
download_db_log_file_portion DownloadDBLogFilePortion
failover_db_cluster FailoverDBCluster
failover_global_cluster FailoverGlobalCluster
list_tags_for_resource ListTagsForResource
modify_activity_stream ModifyActivityStream
modify_certificates ModifyCertificates
modify_current_db_cluster_capacity ModifyCurrentDBClusterCapacity
modify_custom_db_engine_version ModifyCustomDBEngineVersion
modify_db_cluster ModifyDBCluster
modify_db_cluster_endpoint ModifyDBClusterEndpoint
modify_db_cluster_parameter_group ModifyDBClusterParameterGroup
modify_db_cluster_snapshot_attribute ModifyDBClusterSnapshotAttribute
modify_db_instance ModifyDBInstance
modify_db_parameter_group ModifyDBParameterGroup
modify_db_proxy ModifyDBProxy
modify_db_proxy_endpoint ModifyDBProxyEndpoint
modify_db_proxy_target_group ModifyDBProxyTargetGroup
modify_db_snapshot ModifyDBSnapshot
modify_db_snapshot_attribute ModifyDBSnapshotAttribute
modify_db_subnet_group ModifyDBSubnetGroup
modify_event_subscription ModifyEventSubscription
modify_global_cluster ModifyGlobalCluster
modify_option_group ModifyOptionGroup
promote_read_replica PromoteReadReplica
promote_read_replica_db_cluster PromoteReadReplicaDBCluster
purchase_reserved_db_instances_offering PurchaseReservedDBInstancesOffering
reboot_db_cluster RebootDBCluster
reboot_db_instance RebootDBInstance
register_db_proxy_targets RegisterDBProxyTargets
remove_from_global_cluster RemoveFromGlobalCluster
remove_role_from_db_cluster RemoveRoleFromDBCluster
remove_role_from_db_instance RemoveRoleFromDBInstance
remove_source_identifier_from_subscription RemoveSourceIdentifierFromSubscription
remove_tags_from_resource RemoveTagsFromResource
reset_db_cluster_parameter_group ResetDBClusterParameterGroup
reset_db_parameter_group ResetDBParameterGroup
restore_db_cluster_from_s3 RestoreDBClusterFromS3
restore_db_cluster_from_snapshot RestoreDBClusterFromSnapshot
restore_db_cluster_to_point_in_time RestoreDBClusterToPointInTime
restore_db_instance_from_db_snapshot RestoreDBInstanceFromDBSnapshot
restore_db_instance_from_s3 RestoreDBInstanceFromS3
restore_db_instance_to_point_in_time RestoreDBInstanceToPointInTime
revoke_db_security_group_ingress RevokeDBSecurityGroupIngress
start_activity_stream StartActivityStream
start_db_cluster StartDBCluster
start_db_instance StartDBInstance
start_db_instance_automated_backups_replication StartDBInstanceAutomatedBackupsReplication
start_export_task StartExportTask
stop_activity_stream StopActivityStream
stop_db_cluster StopDBCluster
stop_db_instance StopDBInstance
stop_db_instance_automated_backups_replication StopDBInstanceAutomatedBackupsReplication
switchover_read_replica SwitchoverReadReplica
Client has method 'get_caller_identity': False
Client has method 'add_role_to_db_cluster': True
```

Reply:
You mean smth [like this](https://github.com/apache/airflow/pull/27656/commits/1189671f2632572d5a10e4de9c48a0e1a5ebe319), right ?

Reply:
Actually it is just an idea. 
Unfortunately not everything in `boto3` / `botocore` documented 😠 

But yeah... something like that